shared/common-adapters/hoc-timers.js

```diff
@@ -1,33 +1,36 @@
 // @flow
 import * as React from 'react'
 
-type TimerProps = {
-  setTimeout: (func: () => void, timing: number) => TimeoutID,
-  clearTimeout: (id: TimeoutID) => void,
-  setInterval: (func: () => void, timing: number) => IntervalID,
-  clearInterval: (id: IntervalID) => void,
-}
-
-// TODO couldn't figure out a quick way to type this correctly
-// type OptionalProps = {
-// innerRef?: ?(?React.Component<any>) => void,
-// }
+type TimerProps = {|
+  setTimeout: ((func: () => void, timing: number) => TimeoutID) | void,
+  clearTimeout: ((id: TimeoutID) => void) | void,
+  setInterval: ((func: () => void, timing: number) => IntervalID) | void,
+  clearInterval: ((id: IntervalID) => void) | void,
+|}
 
 // Use this to mix your props with timer props like type Props = PropsWithTimer<{foo: number}>
 export type PropsWithTimer<P> = {|
   ...$Exact<P>,
-  ...$Exact<TimerProps>,
+  setTimeout: (func: () => void, timing: number) => TimeoutID,
+  clearTimeout: (id: TimeoutID) => void,
+  setInterval: (func: () => void, timing: number) => IntervalID,
+  clearInterval: (id: IntervalID) => void,
 |}
 
-function getDisplayName(WrappedComponent): string {
-  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
+function getDisplayName(Component): string {
+  return Component.displayName || Component.name || 'Component'
 }
 
-function HOCTimers<Props: TimerProps>(
-  WrappedComponent: React.ComponentType<Props>
-): React.ComponentType<$Diff<Props, TimerProps>> {
-  class TimersComponent extends React.Component<$Diff<Props, TimerProps>> {
-    static displayName = `HOCTimers(${getDisplayName(WrappedComponent)})`
+function hOCTimers<Config: {}, Instance>(
```

Comment:
`HOCTimers`

Reply:
this is a function that returns a class so marking it as such. .matches the style in the flow docs: https://flow.org/en/docs/react/hoc/ and react docs: https://reactjs.org/docs/higher-order-components.html